xchange-coindeal/src/main/java/org/knowm/xchange/coindeal/service/CoindealMarketDataService.java

```diff
@@ -0,0 +1,41 @@
+package org.knowm.xchange.coindeal.service;
+
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.coindeal.CoindealAdapters;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trades;
+import org.knowm.xchange.service.marketdata.MarketDataService;
+import org.knowm.xchange.service.marketdata.params.Params;
+
+import java.io.IOException;
+import java.util.List;
+
+public class CoindealMarketDataService extends CoindealMarketDataServiceRaw
+        implements MarketDataService {
+
+    public CoindealMarketDataService(Exchange exchange) {
+        super(exchange);
+    }
+
+    @Override
+    public Ticker getTicker(CurrencyPair currencyPair, Object... args) throws IOException {
+        return null;
+    }
+
+    @Override
+    public List<Ticker> getTickers(Params params) throws IOException {
+        return null;
+    }
+
+    @Override
+    public OrderBook getOrderBook(CurrencyPair currencyPair, Object... args) throws IOException {
+        return CoindealAdapters.adaptOrderBook(getCoindealOrderbook(currencyPair),currencyPair);
```

Comment:
implementation of methods from generic interfaces should adapt exchange specific exceptions from *Raw methods in generic exceptions

use an *ErrorAdapter for that as described here: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices#step-7